File: variants/java-gradle/airandom/src/main/java/net/pijcke/nim/airandom/RandomAI.java
package net.pijcke.nim.airandom;

import net.pijcke.nim.rules.Nim;
import net.pijcke.nim.rules.Player;

import java.util.Random;

public class RandomAI implements Player {

    private final Random random;

    public RandomAI() {
        this.random = new Random();
    }

    @Override
    public String name() {
        return "Random AI";
    }

    @Override
    public int play(int sticks) {
        return random.nextInt(Math.min(sticks, Nim.MAX_STICKS_PER_TURN));
    }

}

File: variants/kotlin-gradle/nim/airandom/src/main/kotlin/net/pijcke/nim/airandom/RandomAi.kt
package net.pijcke.nim.airandom

import net.pijcke.nim.rules.Nim
import net.pijcke.nim.rules.Player
import java.util.Random
import kotlin.math.min

class RandomAi : Player {

    private val random = Random()

    override fun name(): String = "Random AI"

    override fun play(sticks: Int): Int {
        return random.nextInt(min(Nim.MAX_STICKS_PER_TURN, sticks)) + 1
    }

}